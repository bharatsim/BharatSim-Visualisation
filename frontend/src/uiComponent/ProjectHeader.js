import React from 'react';
import { Box, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useHistory } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import { ChildrenPropTypes } from '../commanPropTypes';

const useProjectLayoutStyle = makeStyles((theme) => {
  return {
    projectNameBar: {
      textTransform: 'capitalize',
      boxShadow: '0px 1px 1px rgba(78, 96, 176, 0.3)',
      height: theme.spacing(12),
      padding: theme.spacing(0, 8),
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
    buttonRoot: {
      padding: 0,
    },
  };
});

function ProjectHeader({ children }) {
  const classes = useProjectLayoutStyle();
  const history = useHistory();

  function openRecentProjects() {
    history.push('/');
  }

  return (
    <Box className={classes.projectNameBar}>
      <Typography variant="h5">{children}</Typography>
      <Button
        onClick={openRecentProjects}
        variant="text"
        size="small"
        classes={{ root: classes.buttonRoot }}
      >
        Back to recent projects
      </Button>
    </Box>
  );
}

ProjectHeader.propTypes = {
  children: ChildrenPropTypes.isRequired,
};

export default ProjectHeader;
