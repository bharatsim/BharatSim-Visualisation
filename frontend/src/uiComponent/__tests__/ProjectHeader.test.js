import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import ProjectHeader from '../ProjectHeader';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<ProjectHeader />', () => {
  it('should match snapshot', () => {
    const { container } = render(<ProjectHeader>Title</ProjectHeader>);

    expect(container).toMatchSnapshot();
  });

  it('should navigate to recent projects on click of back to recent button', () => {
    const { getByText } = render(<ProjectHeader>Title</ProjectHeader>);

    fireEvent.click(getByText('Back to recent projects'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
