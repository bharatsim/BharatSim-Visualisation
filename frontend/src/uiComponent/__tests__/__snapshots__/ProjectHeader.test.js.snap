// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectHeader /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-projectNameBar-1"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Title
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-buttonRoot-2 MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Back to recent projects
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
